### ISSUE TITLE
Implement ``select type`` feature in ASR

### ISSUE THREAD

**Issue by czgdp1807** [lfortran/lfortran#1066]
https://www.intel.com/content/www/us/en/develop/documentation/fortran-compiler-oneapi-dev-guide-and-reference/top/language-reference/a-to-z-reference/s-1/select-type.html

cc: @certik

**Comment by certik** [comment]
Yes, we have to implement it in ASR using some initial design and convert all code from AST->ASR.

Then once we get to implement this in LLVM, we simply iterate on the design, change things as needed.
